### PR TITLE
fix: added support for qrcode widget

### LINKED ISSUE #828: Support for QRCode widget
here is the Kitchen Sink from ensemble-flutter for QRCode widget: https://studio.ensembleui.com/app/e24402cb-75e2-404c-866c-29e6c3dd7992/screen/f4921ba2-9f3d-4c33-8848-4df29a5e7a4d

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
Added support for qrcode widget

## Issue ticket number and link

Closes #828 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

**Comment by sharjeelyunus** [comment]
hey @sagardspeed2, can you please add these properties as well (supported by antd design)
- onRefresh
- status
- icon
- iconSize

File: packages/runtime/src/widgets/QrCode.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+type QRCodeProps = {
+  value: string;
+  size: number;
+  icon: string;
+  iconSize: number;
+  status: "active" | "expired" | "loading" | "scanned";
+  onRefresh: EnsembleAction;
+} & EnsembleWidgetProps;
+
+export const QrCode: React.FC<QRCodeProps> = (props) => {
```
Comment (evshi) [inline packages/runtime/src/widgets/QrCode.tsx:22]: `QRCode`

File: packages/runtime/src/widgets/QrCode.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    rest.id,
+    {
+      setValue: setQrValue,
+    },
+  );
+
+  const onRefreshAction = useEnsembleAction(onRefresh);
+
+  // trigger on signin action
+  const onRefresgActionCallback = useCallback(() => {
```
Comment (evshi) [inline packages/runtime/src/widgets/QrCode.tsx:37]: typo

File: packages/runtime/src/widgets/QrCode.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    if (!onRefreshAction) {
+      return;
+    }
+
+    return onRefreshAction.callback();
+  }, [onRefreshAction]);
+
+  return (
+    <div>
```
Comment (evshi) [inline packages/runtime/src/widgets/QrCode.tsx:46]: Unnecessary div.

File: packages/runtime/src/widgets/QrCode.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+const widgetName = "QRCode";
+
+type QRCodeProps = {
+  value: string;
+  size: number;
+  icon: string;
+  iconSize: number;
+  status: "active" | "expired" | "loading" | "scanned";
+  onRefresh: EnsembleAction;
```
Comment (evshi) [inline packages/runtime/src/widgets/QrCode.tsx:19]: Some of these props are optional?

File: packages/runtime/src/widgets/QrCode.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    if (!onRefreshAction) {
+      return;
+    }
+
+    return onRefreshAction.callback();
+  }, [onRefreshAction]);
+
+  return (
+    <div>
+      <QRCode
```
Comment (evshi) [inline packages/runtime/src/widgets/QrCode.tsx:47]: Need to pass in `rootRef`

File: packages/runtime/src/widgets/QrCode.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        bgColor={values?.styles?.backgroundColor}
+        color={values?.styles?.color}
+        icon={values?.icon}
+        iconSize={values?.iconSize || 40}
+        onRefresh={onRefresgActionCallback}
+        size={values?.size || 160}
+        status={values?.status}
+        value={qrValue || ""}
```
Comment (evshi) [inline packages/runtime/src/widgets/QrCode.tsx:55]: Can we set styles?

**Review by evshi — CHANGES_REQUESTED** [review]

File: packages/runtime/src/widgets/QrCode.tsx (hunk 6 above)
Comment (sagardspeed2) [reply packages/runtime/src/widgets/QrCode.tsx:55]: @evshi The Ant Design `QrCode` widget does not provide direct parameters for styling. However, custom styles can be applied by using the `style` attribute to override the default Ant Design `QrCode` classes.

example
```js
<div ref={rootRef}>
      <style>
      ....
      </style>
      <AntQRCode
        bgColor={values?.styles?.backgroundColor}
        color={values?.styles?.color}
        icon={values?.icon}
        iconSize={values?.iconSize || 40}
        onRefresh={onRefreshActionCallback}
        size={values?.size || 160}
        status={values?.status}
        value={qrValue || ""}
      />
    </div>
```

Comment (evshi) [reply packages/runtime/src/widgets/QrCode.tsx:55]: Hmm okay, this is fine for now then.

File: packages/runtime/src/widgets/QrCode.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  const onRefreshActionCallback = useCallback(() => {
+    if (!onRefreshAction) {
+      return;
+    }
+
+    return onRefreshAction.callback();
+  }, [onRefreshAction]);
+
+  return (
+    <div ref={rootRef}>
```
Comment (evshi) [inline packages/runtime/src/widgets/QrCode.tsx:46]: Is this div necessary? If not, please remove.

**Review by evshi — APPROVED** [review]

File: packages/runtime/src/widgets/QrCode.tsx (hunk 7 above)
Comment (sagardspeed2) [reply packages/runtime/src/widgets/QrCode.tsx:46]: @evshi yes this div is necessary for assign the `ref`, as `QrCode` widget of antd does not support the `ref` prop

**Pull request merged**